### DISCUSSION TITLE
Group mapping

### DISCUSSION THREAD

**Discussion by oussemabhouri** [hkamel/sonar-auth-aad#122] [Q&A]
Hello, 
Is there any way i could map my groups in azure AD to sonarqube? In other terms, users who are in group ADMINS@mycompany.com in AAD must have admin permissions in sonarqube, and developers in ( DEVS@mycompany.com ) must have some specific permissions ( execute analysis and create projects. ) 
Is there any way to do it ? i think it is not something related to the plugin specifically, but i thought this is the best spot to ask my question in. 
thanks in advance :)

**Comment by srvrguy** [comment]
You're looking for the "group sync" feature, which is explained at [https://github.com/hkamel/sonar-auth-aad/wiki/Group-Sync](this page on the wiki). It's fairly straightforward to use.

**Comment by oussemabhouri** [comment]
So I did everything mentioned in the setup and I still get : 


Sorry, but we’re having trouble signing you in.

AADSTS650056: Misconfigured application. This could be due to one of the following: the client has not listed any permissions for 'AAD Graph' in the requested permissions in the client's application registration. Or, the admin has not consented in the tenant. Or, check the application identifier in the request to ensure it matches the configured client application identifier. Or, check the certificate in the request to ensure it's valid. Please contact your admin to fix the configuration or consent on behalf of the tenant. Client app ID: ...

**Comment by oussemabhouri** [comment]
UPDATE: Now I am getting : 
You're not authorized to access this page. Please contact the administrator.

And #116 did not fix the issue for me... 
Also, please note that I am willing to use this plugin mainly for the groupsync feature. It is disabled now, but once i get sso work, will it be fine to activate groupsync? (also pn: I am using this plugin for a single tenant.)

**Comment by oussemabhouri** [comment]
UPDATE:
Everything works as expected. Thanks :)

**Reply by srvrguy** [reply]
Could you provide some details on how you fixed the issue, so others searching might be able to get information that could help them?

**Comment by hdthuong** [comment]
Hello,

When I read the synchronization group installation, I see that in the API permission "Directory.AccessAsUser.All" is required, I don't understand how this can be used for the synchronization of sonarqube groups with the aad. the cybersecurity team is blocking me hard with this api permission . 
After some research would "GroupMember.Read.All" be enough?

best regard

**Reply by srvrguy** [reply]
For getting group membership, we're using "transitiveMemberOf". The permissions that are required to use that endpoint are documented by Microsoft [here](https://docs.microsoft.com/en-us/graph/api/user-list-transitivememberof). Unfortunately, GroupMember.Read.All isn't in that list. If the concern is about write access, potentially Directory.Read.All would work.
